### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by blindpirate**

### CODE REVIEW

File: src/main/kotlin/com/github/blindpirate/GradleEnterpriseHelperPlugin.kt [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+            remoteBuildCache.isPush = isCiServer && remotePush
+            if (remoteCacheUsername.isNotEmpty() && remoteCachePassword.isNotEmpty()) {
+                remoteBuildCache.credentials {
+                    it.username = remoteCacheUsername
+                    it.password = remoteCachePassword
+                }
+            }
+        }
+
+        if (!disableLocalCache) {
```
Comment (wolfs) [inline src/main/kotlin/com/github/blindpirate/GradleEnterpriseHelperPlugin.kt:70]:
```suggestion
        if (disableLocalCache) {
```

I suppose this condition is inverted.

File: src/main/kotlin/com/github/blindpirate/GradleEnterpriseHelperPlugin.kt [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        buildScan.apply {
+            buildScan.server = gradleEnterpriseServerUrl
+            buildScan.isCaptureTaskInputFiles = true
+            buildScan.obfuscation.ipAddresses { addresses: List<InetAddress?> -> addresses.stream().map { address: InetAddress? -> "0.0.0.0" }.collect(Collectors.toList()) }
+            buildScan.publishAlways()
+            (buildScan as BuildScanExtensionWithHiddenFeatures).publishIfAuthenticated()
+            try {
+                buildScan.isUploadInBackground = !isCiServer
+            } catch (ex: NoSuchMethodError) {
+                // GE Plugin version < 3.3. Continue
```
Comment (wolfs) [inline src/main/kotlin/com/github/blindpirate/GradleEnterpriseHelperPlugin.kt:98]: Do we need to support GE plugin versions < 3.3?

File: src/main/kotlin/com/github/blindpirate/GradleEnterpriseHelperPlugin.kt [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+import java.util.stream.Collectors
+
+
+val isCiServer = System.getenv().containsKey("CI")
+val gradleEnterpriseServerUrl = System.getProperty("gradle.enterprise.url") ?: "https://ge.gradle.org"
+val remoteCacheUrl = System.getProperty("gradle.cache.remote.url")?.let { URI(it) } ?: determineCacheNode()
+val remotePush = System.getProperty("gradle.cache.remote.push") != "false"
+val remoteCacheUsername = System.getProperty("gradle.cache.remote.username", "")
+val remoteCachePassword = System.getProperty("gradle.cache.remote.password", "")
+val disableLocalCache = System.getProperty("disableLocalCache")?.toBoolean() ?: false
```
Comment (wolfs) [inline src/main/kotlin/com/github/blindpirate/GradleEnterpriseHelperPlugin.kt:23]: Why is the logic converting `disableLocalCache` to a boolean different to `remotePush`? I think we consistently should use the same logic.

File: README.md [hunk 4]
```diff
@@ -1 +1,103 @@
...
+    - Upload build scans in the foreground.
+  - For local build:
+    - Add `LOCAL` build scan tag.
+    - Add build scan custom value `Git Commit ID` by running `git rev-parse --verify HEAD`.
+    - If running in IDEA:
+      - Add `IDEA` build scan tag.
+      - Add build scan custom value `IDEA version` to IDEA version.
+    - Upload build scans in the background.
+  - For CI and local builds:
+    - Add build scan custom value `Git Branch Name` by running ``.
```
Comment (wolfs) [inline README.md:34]: There is something missing here.

File: README.md [hunk 5]
```diff
@@ -1 +1,103 @@
...
+
+## Use the plugin
+
+The plugin is published to gradle plugin portal.
+
+This is done by configuring a plugin management repository in `settings.gradle`, as shown in the following example:
+
+```
+pluginManagement {
+	repositories {
```
Comment (wolfs) [inline README.md:47]: It seems like this file uses tabs and spaces mixed. Can you add an `.editorconfig` file with a configuration and then use tabs and spaces consistently.

File: README.md [hunk 6]
```diff
@@ -1 +1,103 @@
...
+		gradlePluginPortal()
```
Comment (wolfs) [inline README.md:48]: Why do you need to add the `gradlePluginPortal()` as a repository? Isn't that added automatically?

File: README.md [hunk 7]
```diff
@@ -1 +1,103 @@
...
+	}
+}
+```
+
+
+```
+plugins {
+	// …
+	id "com.gradle.build-scan" version "<<version>>"
+	id "io.spring.gradle-enterprise-conventions" version "<<version>>"
```
Comment (wolfs) [inline README.md:58]: I am pretty sure the id should be different.

File: README.md [hunk 8]
```diff
@@ -1 +1,103 @@
...
+To enable build scan publishing, you need to correctly authenticate as documeneted [here](https://docs.gradle.com/enterprise/gradle-plugin/#authenticating_with_gradle_enterprise).
+
+## Development
+
+Feel free to fork this repository, customize the plugin, and make contribution!
+
+You can install the plugin to local maven repository via:
+
+```
+./gradlew publishPluginMavenPublicationToMavenLocal
```
Comment (wolfs) [inline README.md:80]: Can't you use composite build to test out the plugin? Or does that not work for settings plugins?

File: README.md [hunk 9]
```diff
@@ -1 +1,103 @@
...
+		gradlePluginPortal()
+	}
+}
+```
+
+
+```
+plugins {
+	// …
+	id "com.gradle.build-scan" version "<<version>>"
```
Comment (wolfs) [inline README.md:57]: I suppose this should be the Gradle enterprise plugin, right?

**Comment by blindpirate** [comment]
@wolfs Thanks! I moved the code to https://github.com/gradle/gradle/pull/13597 and fixed issues there.

File: src/main/kotlin/com/github/blindpirate/GradleEnterpriseHelperPlugin.kt (hunk 1 above)
Comment (blindpirate) [reply src/main/kotlin/com/github/blindpirate/GradleEnterpriseHelperPlugin.kt:70]: Yes, thanks!

File: src/main/kotlin/com/github/blindpirate/GradleEnterpriseHelperPlugin.kt (hunk 2 above)
Comment (blindpirate) [reply src/main/kotlin/com/github/blindpirate/GradleEnterpriseHelperPlugin.kt:98]: I'm not sure. We expect this plugin to be a sample for GE users, who might be using < 3.3 - this is my assumption.

File: src/main/kotlin/com/github/blindpirate/GradleEnterpriseHelperPlugin.kt (hunk 3 above)
Comment (blindpirate) [reply src/main/kotlin/com/github/blindpirate/GradleEnterpriseHelperPlugin.kt:23]: The logic was from here: https://github.com/gradle/gradle/blob/master/gradle/build-cache-configuration.settings.gradle.kts#L27-L32

But this is not an excuse, we should use the same.

File: README.md (hunk 9 above)
Comment (blindpirate) [reply README.md:57]: yes.

File: README.md (hunk 8 above)
Comment (blindpirate) [reply README.md:80]: I assume settings plugin happens very early, though I should give it a try.

File: README.md (hunk 7 above)
Comment (blindpirate) [reply README.md:58]: Fixed.

**Pull request closed without merging**